socketio/src/client/client.rs

```diff
@@ -468,6 +568,53 @@ mod test {
         Ok(())
     }
 
+    #[test]
+    fn socket_io_reconnect_integration() -> Result<()> {
+        let url = crate::test::socket_io_restart_server();
+
+        let socket = ClientBuilder::new(url).connect();
+        assert!(socket.is_ok(), "should connect success");
+        let socket = socket.unwrap();
+
+        std::thread::sleep(std::time::Duration::from_millis(200));
```

Comment:
nit: why sleep?

Reply:
Good catch, I found connect function return not means connect success, the first emit message will not success if not wait.

Reply:
blocker: That issue doesn't exist in main, connect should block until the client is usable.